src/lib.rs

```diff
@@ -424,13 +444,16 @@ where
     {
         for p in pixels {
             let Pixel(coord, color) = p;
-            let x = coord.x as u8;
-            let y = coord.y as u8;
-            let c = match color {
-                BinaryColor::Off => 0,
-                BinaryColor::On => 1,
-            };
-            self.set_pixel(x, y, c);
+            if coord.x >= 0 && coord.x < WIDTH as i32 &&
```

Comment:
Is the check here needed? The same is also done in set_pixel.

Reply:
Yes, well the problem here is, that we have a type conversion from i32 to u8 going on here. In set_pixel only the u8 bounds are checked. If we don't check bounds before cast, funny things will happen, if the cast cuts out the low byte from the value. Even if we have a check in set_pixel that can cause unwanted misbehavior, as the lower byte may be in bounds, while the whole i32 coordinate was out of bounds.

Reply:
yeah, right